test/integration/array_test.ts

```diff
@@ -194,6 +221,47 @@ describe('Array', function () {
     }, this.test!.title);
   });
 
+  it('Can handle concurrent moveFront operations', async function () {
+    await withTwoClientsAndDocuments(async (c1, d1, c2, d2) => {
+      d1.update((root) => {
+        root['k1'] = [0, 1, 2];
+        assert.equal('{"k1":[0,1,2]}', root.toJSON());
+      });
+      await c1.sync();
+      await c2.sync();
+      assert.equal(d1.toJSON(), d2.toJSON());
+
+      d1.update((root) => {
+        const item = root['k1'].getElementByIndex(2);
+        root['k1'].moveFront(item.getID());
+        assert.equal('{"k1":[2,0,1]}', root.toJSON());
+      });
+
+      d1.update((root) => {
+        const item = root['k1'].getElementByIndex(2);
+        root['k1'].moveFront(item.getID());
+        assert.equal('{"k1":[1,2,0]}', root.toJSON());
+      });
+
+      d2.update((root) => {
+        const item = root['k1'].getElementByIndex(2);
+        root['k1'].moveFront(item.getID());
+        assert.equal('{"k1":[2,0,1]}', root.toJSON());
+      });
+
+      d2.update((root) => {
+        const item = root['k1'].getElementByIndex(2);
+        root['k1'].moveFront(item.getID());
+        assert.equal('{"k1":[1,2,0]}', root.toJSON());
+      });
+
+      await c1.sync();
+      await c2.sync();
+      await c1.sync();
+      assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
+    }, this.test!.title);
+  });
```

Comment:
When I simplify the test to the following, the synchronization fails. It would be good to check the reason for the divergence.

```suggestion
it.only('Can handle concurrent moveFront operations', async function () {
  await withTwoClientsAndDocuments(async (c1, d1, c2, d2) => {
    d1.update((root) => {
      root['k1'] = [0, 1, 2];
      assert.equal('{"k1":[0,1,2]}', root.toJSON());
    });
    await c1.sync();
    await c2.sync();
    assert.equal(d1.toJSON(), d2.toJSON());

    d1.update((root) => {
      const item = root['k1'].getElementByIndex(2);
      root['k1'].moveFront(item.getID());
      assert.equal('{"k1":[2,0,1]}', root.toJSON());
    });

    d2.update((root) => {
      const item = root['k1'].getElementByIndex(1);
      root['k1'].moveFront(item.getID());
      assert.equal('{"k1":[1,0,2]}', root.toJSON());
    });

    await c1.sync();
    await c2.sync();
    await c1.sync();
    assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
  }, this.test!.title);
});
```